src/core/lib/compression/message_compress.cc

```diff
@@ -149,7 +149,13 @@ static int copy(grpc_slice_buffer* input, grpc_slice_buffer* output) {
 }
 
 static int compress_inner(grpc_message_compression_algorithm algorithm,
-                          grpc_slice_buffer* input, grpc_slice_buffer* output) {
+                          grpc_slice_buffer* input, grpc_slice_buffer* output)
+#if defined(__clang__)
```

Comment:
We like to avoid this compiler specific annotation as much as possible. Could you achieve this without using this like adding default section to switch?

Reply:
If I'm remembering right, there is a test to see what happens when an enum is passed in which is not in the list of accepted enums.  Per the C++ standard, this is undefined behavior, so Clang's ubsan detects it and complains.

I spent some time looking around and the best advice I found was to look for the presence of the __clang__ define.  Do you have any better ideas?

(I'm happy to drop any pieces of this you want me to as well.  This is what we ran)

Reply:
How about this?

```
  switch (algorithm) {
    case GRPC_MESSAGE_COMPRESS_NONE:
      /* the fallback path always needs to be send uncompressed: we simply
         rely on that here */
      return 0;
    case GRPC_MESSAGE_COMPRESS_DEFLATE:
      return zlib_compress(input, output, 0);
    case GRPC_MESSAGE_COMPRESS_GZIP:
      return zlib_compress(input, output, 1);
    case GRPC_MESSAGE_COMPRESS_ALGORITHMS_COUNT:
      break;
   default: // added here
     break;
  }
```

Reply:
That is undefined behavior in C++.  @bsilver8192 knows a lot more about this part of the C++ standard.  He also didn't have any better ideas than looking for __clang__

https://godbolt.org/z/GGYo7z shows a stripped down example.  Adding a default is still undefined behavior.

Reply:
I can't remember if I said this earlier, but there is a test which tests what happens if an unknown enum gets passed in.  That is triggering ubsan (rightfully so)

Reply:
I think [test/core/util/ubsan_suppressions.txt](https://github.com/grpc/grpc/blob/b669a3c521809c9b193a8f39b7d9195014b3f19c/test/core/util/ubsan_suppressions.txt#L7) covers all of these, and dropping these parts of the patch makes the most sense.

About why default doesn't help: pre-C++17 the conversion gives an unspecified result, and C++17 downgrades it to undefined behavior. The problem is the underlying type of the enum might only have enough bits to hold all the values in the enum (because it's a non-`enum class` without a specified underlying type), which means converting any values that require more bits to represent isn't well-defined.

Reply:
Oh understood. Thank you for the context. 

As @bsilver8192, these warning are disabled by test/core/util/ubsan_suppressions.txt, removing these would make more sense.

Reply:
Patch reverted, please take a look!

Reply:
Looks great! thank you!